### ISSUE TITLE
FOXX Authentication documentation is fragmented, inconsistent and wrong

### ISSUE THREAD

**Issue by jakobrosenberg** [arangodb/arangodb#1926]
I'm having a rough time figuring how to implement user authentication.

I read that **Oauth** is implemented and working directly in Foxx 3, but it's undocumented. 
**Local authentication** is documented, but not working.

https://docs.arangodb.com/devel/Manual/Foxx/Auth.html
```
const sessions = sessionsMiddleware({
  storage: module.context.collection('sessions'),
  transports: 'cookie'
});
```

https://docs.arangodb.com/devel/Manual/Foxx/Users.html
```
const sessions = sessionsMiddleware({
  storage: module.context.collection('sessions'),
  transports: ['cookie']
});
```

Both of the above produce the error. ``AssertionError: Must specify at least one session transport``

The solution is found here
https://docs.arangodb.com/devel/Manual/Foxx/Migrating2x/Sessions.html
```
router.use(sessionMiddleware({
  storage: module.context.collection('sessions'),
  transport: cookieTransport('keyboardcat')
}));
```

The next error is found at ``router.post('/login'...``. Here ``req.bodyParams.username`` has to be replaced with ``req.body.username`` while ``user.authData`` is undefined. Changing ``user.authData`` to ``user.password`` causes auth.verify to throw an error. There's no mention of what values auth.verify accepts so I'm at a loss.

Are the user/authentication documents meant for v2.x? I feel uncomfortable implementing security using broken guides.

Hopefully I missed something and the error is on my end.

**Comment by pluma** [comment]
I can reassure you that it's the documentation that is wrong rather than the implementation.

The arguments to `sessionMiddleware` are in singular form. This changed throughout the development of 3.0 and some pieces of the documentation still seem to refer to the plural form. However the middleware *should* log warnings to the ArangoDB logs if you use the plural form to aid you in debugging this.

The `verify` method takes a hash object and a password string. The hash object is the output of the `create` method, i.e. a plain object with the three properties listed in the documentation. The users documentation incorrectly gives a signup example that just stores the password. It should `create` the hash object instead and store it in a property called `authData` and delete the password from the user object before saving it.

You're right that these examples shouldn't be broken. Thanks for reporting these mistakes so we can fix them.

As for security: the "simple auth" and "oauth2" implementations in ArangoDB 3.0 are direct ports of the Foxx standalone services that have existed for a long time throughout 2.x. The underlying implementation hasn't changed. So while the documentation may contain minor errors like misspellings or omitting steps, the implementation is far more mature.

**Comment by pluma** [comment]
Fixed via 9c4320a.
